### ISSUE TITLE
inactive members are being sent to the rfid reader

### ISSUE THREAD

**Issue by dfirebaugh** [HackRVA/memberdashboard#104]
An inactive user is being sent to the rfid reader.

- I log into the rfid reader and manually delete the user.
- I verify that the user was removed from the device
- a day later, I see that user is added back to the rfid reader.

At the moment, I'm not sure if the issue is on the server or on the rfid reader.

**Comment by Ranthalion** [comment]
@dfirebaugh Questions:

- Were you able to confirm that the member was marked inactive in the DB?  
- Did the member change back to active in the db at some point?
- Any recommendations on emulating the rfid in a docker container?

**Comment by dfirebaugh** [comment]
i tracked this down to this query 
https://github.com/HackRVA/memberdashboard/blob/b2a3762f543b77c1693032e780de6815d18233dd/database/resourcedbmethod.go#L78-L86

Which this was triggering daily via the scheduler https://github.com/HackRVA/memberdashboard/blob/b2a3762f543b77c1693032e780de6815d18233dd/resourcemanager/resourcemanager.go#L97

It needs an additional condition to check the persons membership level.

I was investigating ways to test the db before implementing something.

Maybe something like [ramsql](https://github.com/proullon/ramsql)










To answer your questions:
* the member was inactive in the db
* the member was inactive in the db
* there are some scripts in the resourcedummy folder https://github.com/HackRVA/memberdashboard/blob/main/test/resourcedummy
basically you would probably download an mqtt client and subscribe the the topic.  Or we could build out that resource dummy, but that's probably not all that valuable.

**Comment by Ranthalion** [comment]
That looks like a pretty easy fix, but I agree with where you are going with the integration test.  ramsql might be a good choice, especially for unit testing logic.  Docker containers could make it pretty easy to set up real automated integration tests.  I won't have any time to work on it in July, but I may be able to try something in August if it's still a goal.

**Comment by dfirebaugh** [comment]
cool, I went ahead and pushed an update to that query.

I'll open a new issue for the integration tests.  I don't think there's a rush.
